tests/integration/mirrorpeersecret_controller_test.go

```diff
@@ -0,0 +1,534 @@
+		}
+		managedClusters = append(managedClusters, managedCluster)
+	}
+	return managedClusters
+}
+
+func (mppp MirrorPeerPlusPlus) GetClusterNamespaces() []corev1.Namespace {
+	uniqueNamespaceMap := map[string]struct{}{}
+	for _, eachPeer := range mppp.Spec.Items {
+		uniqueNamespaceMap[eachPeer.ClusterName] = struct{}{}
+	}
+	var namespaceArr []corev1.Namespace
+	for namespace := range uniqueNamespaceMap {
+		namespace := corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+		namespaceArr = append(namespaceArr, namespace)
+	}
+	return namespaceArr
+}
+
+func (mppp MirrorPeerPlusPlus) GetDummySourceSecrets(prefixName string) []corev1.Secret {
+	var destSecrets []corev1.Secret
+	for i, eachPR := range mppp.Spec.Items {
+		secretNN := types.NamespacedName{
+			Name:      fmt.Sprintf("%s_%d", prefixName, i),
+			Namespace: eachPR.ClusterName,
+		}
+		storageClusterNN := types.NamespacedName{
+			Name:      eachPR.StorageClusterRef.Name,
+			Namespace: eachPR.StorageClusterRef.Namespace,
+		}
+		destSecret := common.CreateSourceSecret(secretNN, storageClusterNN, []byte("MySecretData1234"))
+		destSecrets = append(destSecrets, *destSecret)
+	}
+	return destSecrets
+}
+
+func createMirrorPeer(mirrorPeerName string, items ...multiclusterv1alpha1.PeerRef) multiclusterv1alpha1.MirrorPeer {
+	mirrorPeer := multiclusterv1alpha1.MirrorPeer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: mirrorPeerName,
+		},
+		Spec: multiclusterv1alpha1.MirrorPeerSpec{
+			Items: items,
+		},
+	}
+	return mirrorPeer
+}
+
+func createDefaultMirrorPeer() multiclusterv1alpha1.MirrorPeer {
+	defaultMirrorPeerName := "mirrorpeer"
+	items := []multiclusterv1alpha1.PeerRef{
+		{
+			ClusterName: "cluster1",
+			StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+				Name:      "test-storagecluster-1",
+				Namespace: "test-namespace",
+			},
+		},
+		{
+			ClusterName: "cluster2",
+			StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+				Name:      "test-storagecluster-2",
+				Namespace: "test-namespace",
+			},
+		},
+	}
+	return createMirrorPeer(defaultMirrorPeerName, items...)
+}
+
+func createFakeScheme() *runtime.Scheme {
+	// groupVersion is group version used to register these objects
+	groupVersion := rtscheme.GroupVersion{Group: "orchestrator.odf.openshift.io", Version: "v1"}
+	// schemeBuilder is used to add go types to the GroupVersionKind scheme
+	schemeBuilder := &pkgscheme.Builder{GroupVersion: groupVersion}
+
+	scheme, err := schemeBuilder.Build()
+	utilruntime.Must(err)
+
+	utilruntime.Must(schemeBuilder.AddToScheme(scheme))
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(clusterv1.AddToScheme(scheme))
+	utilruntime.Must(multiclusterv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(addonapiv1alpha1.AddToScheme(scheme))
+
+	return scheme
+}
+
+func createFakeClient() client.Client {
+	scheme := createFakeScheme()
+	fc := fake.NewClientBuilder().
+		WithScheme(scheme).Build()
+	return fc
+}
```

Comment:
We have a client created by default. Why do we need this?

Reply:
Unable to use the `k8sClient` created in the `suite_test` as the tests are failing/crashing...